packages/pytest-simcore/src/pytest_simcore/postgres_service.py

```diff
@@ -7,43 +7,46 @@
 
 import pytest
 import sqlalchemy as sa
-import tenacity
 from sqlalchemy.orm import sessionmaker
+from tenacity import Retrying
 
 from servicelib.aiopg_utils import DSN, PostgresRetryPolicyUponInitialization
 from simcore_postgres_database.models.base import metadata
-from utils_docker import get_service_published_port
+
+from .helpers.utils_docker import get_service_published_port
 
 
 @pytest.fixture(scope="module")
 def postgres_dsn(docker_stack: Dict, devel_environ: Dict) -> Dict[str, str]:
     assert "simcore_postgres" in docker_stack["services"]
 
-    DSN = {
+    pg_config = {
         "user": devel_environ["POSTGRES_USER"],
         "password": devel_environ["POSTGRES_PASSWORD"],
         "database": devel_environ["POSTGRES_DB"],
         "host": "127.0.0.1",
         "port": get_service_published_port("postgres", devel_environ["POSTGRES_PORT"]),
     }
     # nodeports takes its configuration from env variables
-    os.environ["POSTGRES_ENDPOINT"] = f"127.0.0.1:{DSN['port']}"
+    os.environ["POSTGRES_ENDPOINT"] = f"127.0.0.1:{pg_config['port']}"
     os.environ["POSTGRES_USER"] = devel_environ["POSTGRES_USER"]
     os.environ["POSTGRES_PASSWORD"] = devel_environ["POSTGRES_PASSWORD"]
     os.environ["POSTGRES_DB"] = devel_environ["POSTGRES_DB"]
-    return DSN
+    return pg_config
 
 
 @pytest.fixture(scope="module")
 def postgres_db(postgres_dsn: Dict[str, str], docker_stack: Dict) -> sa.engine.Engine:
     url = DSN.format(**postgres_dsn)
 
-    # NOTE: Comment this to avoid postgres_service
-    assert wait_till_postgres_responsive(url)
+    # Attempts until responsive
+    for attempt in Retrying(**PostgresRetryPolicyUponInitialization().kwargs):

```

Comment:
nice one, did not know about that Retrying. what is the rationale between this and having a separate function for checking responsiveness?